charts/argo-cd/values.yaml

```diff
@@ -1,76 +1,459 @@
-## Ref: https://github.com/argoproj/argo-cd
-##
-# -- Provide a name in place of `argocd`
-nameOverride: argocd
-# -- String to fully override `"argo-cd.fullname"`
-fullnameOverride: ""
-# -- Override the Kubernetes version, which is used to evaluate certain manifests
-kubeVersionOverride: ""
-
+## Globally shared configuration
 global:
-  skipCrds: true 
-
+  # -- Common labels for the all resources
+  additionalLabels: {}
+    # app: argo-cd 
+  cluster: 
+    ngcp: true
+  region: "LAS1"
+  network:
+    disableK8sNetworking: false
+    vpcs: []
+    # - name: las1-1
+  affinity: 
+    nodeAffinity:
+      requiredDuringSchedulingIgnoredDuringExecution:
+        nodeSelectorTerms:
+          - matchExpressions:
+              - key: topology.kubernetes.io/region
+                operator: In
+                values:
+                  - "{{ .Values.global.region }}"
+              - key: node.coreweave.cloud/class
+                operator: In
+                values:
+                  - cpu
+  # -- Number of old deployment ReplicaSets to retain. The rest will be garbage collected.
+  revisionHistoryLimit: 3
+
+  # Default image used by all components
   image:
     # -- If defined, a repository applied to all Argo CD deployments
     repository: quay.io/argoproj/argocd
     # -- Overrides the global Argo CD image tag whose default is the chart appVersion
     tag: ""
     # -- If defined, a imagePullPolicy applied to all Argo CD deployments
     imagePullPolicy: IfNotPresent
+
+  # -- Secrets with credentials to pull images from a private registry
+  imagePullSecrets: []
+
+  # Default logging options used by all components
+  logging:
+    # -- Set the global logging format. Either: `text` or `json`
+    format: text
+    # -- Set the global logging level. One of: `debug`, `info`, `warn` or `error`
+    level: info
+
+  # -- Annotations for the all deployed Statefulsets
+  statefulsetAnnotations: {}
+
+  # -- Annotations for the all deployed Deployments
+  deploymentAnnotations: {}
+
   # -- Annotations for the all deployed pods
   podAnnotations: {}
+
   # -- Labels for the all deployed pods
   podLabels: {}
-  # -- Toggle and define securityContext. See [values.yaml]
+
+  # -- Toggle and define pod-level security context.
+  # @default -- `{}` (See [values.yaml])
   securityContext: {}
   #  runAsUser: 999
   #  runAsGroup: 999
   #  fsGroup: 999
 
-  # -- If defined, uses a Secret to pull an image from a private Docker registry or repository
-  imagePullSecrets: []
   # -- Mapping between IP and hostnames that will be injected as entries in the pod's hosts files
   hostAliases: []
   # - ip: 10.20.30.40
   #   hostnames:
   #   - git.myhostname
 
-  # -- Additional labels to add to all resources
-  additionalLabels: {}
-    # app: argo-cd
-
   networkPolicy:
     # -- Create NetworkPolicy objects for all components
     create: false
     # -- Default deny all ingress traffic
     defaultDenyIngress: false
 
+region: ""
```

Comment:
duplicate?

Reply:
Nah, just non global override. If a parent chart specifies a certain region you always want the child to be able to modify it. That said I do notice one error which is in the affinity where the retemplating should include a default of that region value.
`  affinity: 
    nodeAffinity:
      requiredDuringSchedulingIgnoredDuringExecution:
        nodeSelectorTerms:
          - matchExpressions:
              - key: topology.kubernetes.io/region
                operator: In
                values:
                  - "{{ default  .Values.global.region .Values.region }}"
              - key: node.coreweave.cloud/class
                operator: In
                values:
                  - cpu`